### ISSUE TITLE
When using an @import (reference), mixins that contain an & selector get added to the compiled output improperly

### ISSUE THREAD

**Issue by bcullman** [less/less.js#1968]
When I import a less file with the (reference) directive, I expect to see no CSS output from that file. 

However, I am seeing content being output from the imported file when a mixin contains an & selector.

__mixin.less__
```less
.mixin-no-parent-selector() {
  background-color: red;
}
.mixin-with-parent-selector() {
  &:first-child{ //  USING AN & SELECTOR HERE
    background-color: blue;
 }
}
```
__less-for-reference.less__
```less
.container-1 {
   .mixin-no-parent-selector();
}
.container-2{
   .mixin-with-parent-selector()
}
```

__style.less__
```less
/*!
 * styles.less
 */
@import "mixin.less";
@import (reference) "less-for-reference.less";
```

__Expected Output__
```css
/*!
 * styles.less
 */
```

__Actual Output__
```css
/*!
 * styles.less
 */

.container-2:first-child{
    background-color: blue;
}
```

**Comment by assadtony** [comment]
Any updates on this issue? :)

**Comment by thybzi** [comment]
Wish that to be fixed. My `.clearfix` mixin runs many-many times on the same selector, because it contains `&:after`

**Comment by seven-phases-max** [comment]
@thybzi 

If it's really your own mixin you can easily make yourself to never suffer from this issue at all.
Simply don't use Bootstrap-like `.clearfix` code. It's actually inefficient and better to be written in opposite:

```
.clearfix_ {
    // actual clearfix code
   &:after { ...}
   // etc.
}

.clearfix() {
    &:extend(.clearfix_ all);
}
```

This way you don't need any [`reference` kludges] and thus never suffer from any of its issues.

**Comment by SomMeri** [comment]
I am working on import reference now, but I am not able to give out estimate.

**Comment by SomMeri** [comment]
This should be fixed by #2729 .
